plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/JadeSchemaGenerator.scala

```diff
@@ -56,13 +68,59 @@ object JadeSchemaGenerator {
     *
     * @param tables collection of tables to include in the schema
     */
-  def generateSchema(tables: Seq[MonsterTable]): Either[String, JadeSchema] =
-    extractRelationships(tables).map { rels =>
+  def generateSchema(
+    tables: Seq[MonsterTable],
+    fragments: Seq[MonsterTableFragment]
+  ): Either[String, JadeSchema] = {
+    val fragmentMap = fragments.map(t => t.name -> t).toMap
+
+    val base: Either[String, List[MonsterTable]] = Right(Nil)
+    val flattenedTables = tables.foldLeft(base) { (acc, table) =>
+      acc.flatMap(tail => flattenFragments(table, fragmentMap).map(_ :: tail))
+    }
+
+    for {
+      tables <- flattenedTables
+      relationships <- extractRelationships(tables)
+    } yield {
       JadeSchema(
         tables = tables.map(convertTable).toSet,
-        relationships = rels.toSet
+        relationships = relationships.toSet
       )
     }
+  }
+
+  private def flattenFragments(
+    table: MonsterTable,
+    fragments: Map[JadeIdentifier, MonsterTableFragment]
+  ): Either[String, MonsterTable] = {
+    val base: Either[String, MonsterTable] = Right(table.copy(tableFragments = Vector.empty))
+    table.tableFragments.foldLeft(base) { (acc, fragmentId) =>
+      acc.flatMap { baseTable =>
+        fragments.get(fragmentId).toRight(s"No such fragment: $fragmentId").flatMap { subTable =>
```

Comment:
I'm going to extract this validation logic into its own method, and wire it up to `sbt test` in the plugin.